internal/controller/vrg_volrep.go

```diff
@@ -1483,13 +1483,17 @@ func (v *VRGInstance) validateVRValidatedStatus(
 	volRep *volrep.VolumeReplication,
 ) (bool, conditionState) {
 	conditionMet, condState, errorMsg := isVRConditionMet(volRep, volrep.ConditionValidated, metav1.ConditionTrue)
-	if !conditionMet && condState != conditionMissing {
-		defaultMsg := "VolumeReplication resource not validated"
-		v.updatePVCDataReadyConditionHelper(pvc.Namespace, pvc.Name, VRGConditionReasonError, errorMsg,
-			defaultMsg)
-		v.updatePVCDataProtectedConditionHelper(pvc.Namespace, pvc.Name, VRGConditionReasonError, errorMsg,
-			defaultMsg)
-		v.log.Info(fmt.Sprintf("%s (VolRep: %s/%s)", defaultMsg, volRep.GetName(), volRep.GetNamespace()))
+	if !conditionMet {
+		if errorMsg == "" {
+			errorMsg = "VolumeReplication resource not validated"
+		}
+		// The condition does not exist when using csi-addons < 0.10.0, so we cannot treat this as a failure.
+		if condState != conditionMissing {
+			v.updatePVCDataReadyCondition(pvc.Namespace, pvc.Name, VRGConditionReasonError, errorMsg)
+			v.updatePVCDataProtectedCondition(pvc.Namespace, pvc.Name, VRGConditionReasonError, errorMsg)
+		}
+
```

Comment:
When csi-addons < 0.10.0 it is not conditionMissing, it is message missing. Previously when condition was missing, we were updating both dataready and dataprotected with error message 
`		errorMsg := fmt.Sprintf("Failed to get the %s condition from status of VolumeReplication resource.",
			conditionType)
`

Reply:
csi-addons added the Validated condition in 0.10.0. The options are:

- csi-addons < 0.10.0:
  - condition is not found
  - don't set the conditions
  - log "Failed to get the condition"
- csi-addons == 0.10.0:
  - condition is found
  - message is empty
  - set the conditions using stale, unknown or the default message: "VolumeReplication resource not validated"
  - log the default message
- csi-addons > 0.10.0:
  - condition found
  - message non-empty
  - we set the conditions using the message from the Validated condition
  - log the same message

What is the issue?

Reply:
Yes, my misunderstanding. I didn't notice, that we are addressing the new condition type, and thought that in case of missing condition we need to print error